web/src/main/java/com/epam/rft/atsy/web/exceptionhandling/UncheckedExceptionResolver.java

```diff
@@ -0,0 +1,82 @@
+package com.epam.rft.atsy.web.exceptionhandling;
+
+import com.epam.rft.atsy.service.exception.DuplicateRecordException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.Ordered;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.json.MappingJackson2JsonView;
+
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This class handles the unchecked exceptions thrown by the controllers. Registered with the
+ * highest precedence among the handlers, it renders a generic error page upon receiving an ordinary
+ * HTTP request or sends a JSON response when it detects an AJAX request.
+ *
+ * Maintains a list of exception classes that require a more specific handling and therefore should
+ * be passed to resolvers with lower precedence.
+ */
+@Component
+public class UncheckedExceptionResolver implements HandlerExceptionResolver, Ordered {
+  private static final ModelAndView PASS_TO_NEXT_RESOLVER = null;
+
+  private static final String ERROR_VIEW_NAME = "error";
+
+  @Autowired
+  private MappingJackson2JsonView jsonView;
+
+  /**
+   * The list of exceptions that should not be handled but passed.
+   */
+  private final List<Class<? extends RuntimeException>> skippedExceptionList = Arrays.asList(
```

Comment:
Modify the order of the exception handlers by overriding the configureExceptionResolvers method in WebConfiguration class.
In that method you can register the exception resolvers in any order.